File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {

        Map<Character, Integer> map = new HashMap<>();
        int result = 0;
        int currVal = 0;
        int length = s.length();
        int index = length - 1;

        map.put('I',1);
        map.put('V',5);
        map.put('X',10);
        map.put('L',50);
        map.put('C',100);
        map.put('D',500);
        map.put('M',1000);

        while(index >= 0)
        {
            int num1 = map.get(s.charAt(index));
            currVal = num1;

            while(index - 1 >=0 && map.get(s.charAt(index - 1)) <  num1)
            {
                currVal = currVal - map.get(s.charAt(index - 1));
                index--;
            }

            result += currVal;
            index--;
        }

        return result;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {

    let result = 0;
    let map = new Map([
        ['I',1],
        ['V',5],
        ['X',10],
        ['L',50],
        ['C',100],
        ['D',500],
        ['M',1000]
     ]);

    result = map.get(s.charAt(s.length - 1));

    for(let i = s.length - 2 ; i >= 0 ; i--)
    {
        
        if(map.get(s.charAt(i)) < map.get(s.charAt(i+1)))
            result-=map.get(s.charAt(i));
        else
            result+=map.get(s.charAt(i));            

    }

    return result;
};